File: 2497-maximum-matching-of-players-with-trainers/maximum-matching-of-players-with-trainers.cpp
class Solution {
public:
    int matchPlayersAndTrainers(vector<int>& players, vector<int>& trainers) {
        sort(players.begin(), players.end());
        sort(trainers.begin(), trainers.end());

        int result = 0;
        for (int i = 0, j = 0, ilen = players.size(), jlen = trainers.size(); i < ilen && j < jlen; i++, j++) {
            if (players[i] <= trainers[j]) {
                result++;
            } else if (players[i] > trainers[j]) {
                i--;
            }
        }

        return result;
        
    }
};

File: 2497-maximum-matching-of-players-with-trainers/maximum-matching-of-players-with-trainers.java
class Solution {
    public int matchPlayersAndTrainers(int[] players, int[] trainers) {
        Arrays.sort(players);
        Arrays.sort(trainers);

        int i = 0, j = 0, result = 0;
        int ilen = players.length;
        int jlen = trainers.length;
        for (; i < ilen && j < jlen; i++, j++) {
            if (players[i] <= trainers[j])
                result++;
            else 
                i--;
        }

        return result;
    }
}